### PR TITLE
Use the proper publish task for nightlies

### PR DESCRIPTION
**Pull request by ambv**
Also, apply all pending changes from templates to workflow files.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -513,7 +513,7 @@ jobs:
         printf 'FROM scratch\nENV EDGEDB_USER edgedb\nADD edgedb /usr/bin/\nENTRYPOINT ["/usr/bin/edgedb"]\n' \
           >dockerctx/Dockerfile
         tag=$(date +%Y%m%d%H%M%S)$(echo "${GITHUB_SHA}" | cut -c1-6)
-        echo "SNAPSHOT_TAG=${tag}" >> $GITHUB_ENV
+        echo ::set-env name=SNAPSHOT_TAG::${tag}
```
Comment (elprans) [inline .github/workflows/nightly.yml:516]: The other version is correct. ::set-env is deprecated for security reasons

Comment (ambv) [reply .github/workflows/nightly.yml:516]: Interesting. I only compiled what's in the templates so I'll be changing it then.

File: .github/workflows.src/nightly.tpl.yml [hunk 2]
```diff
@@ -193,7 +193,7 @@ jobs:
         printf 'FROM scratch\nENV EDGEDB_USER edgedb\nADD edgedb /usr/bin/\nENTRYPOINT ["/usr/bin/edgedb"]\n' \
           >dockerctx/Dockerfile
         tag=$(date +%Y%m%d%H%M%S)$(echo "${GITHUB_SHA}" | cut -c1-6)
-        echo ::set-env name=SNAPSHOT_TAG::${tag}
+        echo "SNAPSHOT_TAG=${tag}" >> $GITHUB_ENV
```
Comment (ambv) [inline .github/workflows.src/nightly.tpl.yml:196]: See previous conversation in this PR. This version is preferred, the other is deprecated as insecure. The rendered workflow files were already updated.

File: .github/workflows.src/release.tpl.yml [hunk 3]
```diff
@@ -144,7 +144,7 @@ jobs:
         printf 'FROM scratch\nENV EDGEDB_USER edgedb\nADD edgedb /usr/bin/\nENTRYPOINT ["/usr/bin/edgedb"]\n' \
           >dockerctx/Dockerfile
         ver=$(dockerctx/edgedb --version | cut -f2 -d' ')
-        echo ::set-env name=CLI_VERSION::${ver}
+        echo "CLI_VERSION=${ver}" >> $GITHUB_ENV
```
Comment (ambv) [inline .github/workflows.src/release.tpl.yml:147]: Ditto.

**Review by elprans — APPROVED** [review]

**Pull request merged**